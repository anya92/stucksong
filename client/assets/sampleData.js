export const recentlyPlayedTracks = [
  {
    album: 'Americana',
    artist: 'The Offspring',
    id: '4EchqUKQ3qAQuRNKmeIpnf',
    image: 'https://i.scdn.co/image/c77a1f3a2de72c2a8fde481a016376ca8ef81b33',
    played_at: '2018-08-14T18:50:09.757Z',
    title: 'The Kids Aren\'t Alright',
    uri: 'spotify:track:4EchqUKQ3qAQuRNKmeIpnf',
  },
  {
    album: 'No Pads, No Helmets...Just Balls (15th Anniversary Tour Edition)',
    artist: 'Simple Plan',
    id: '6otiaV2fagE3s8IvP6WkwG',
    image: 'https://i.scdn.co/image/9f6a521519080242f2de1c2b8873eec5372f7a6a',
    played_at: '2018-08-14T18:46:04.608Z',
    title: 'I\'m Just A Kid',
    uri: 'spotify:track:6otiaV2fagE3s8IvP6WkwG',
  },
  {
    album: 'All Killer, No Filler',
    artist: 'Sum 41',
    id: '1HNE2PX70ztbEl6MLxrpNL',
    image: 'https://i.scdn.co/image/4fc4f0bb924338f85e0fe7c899cde32ec5e9f10d',
    played_at: '2018-08-14T18:42:32.417Z',
    title: 'In Too Deep',
    uri: 'spotify:track:1HNE2PX70ztbEl6MLxrpNL',
  },
];

export const topArtists = [
  {
    genres: [
      'dance pop',
      'folk-pop',
      'indie pop',
      'indie r&b',
      'indietronica',
      'metropopolis',
      'pop',
    ],
    id: '4Ui2kfOqGujY81UcPrb5KE',
    image: 'https://i.scdn.co/image/a20acc6267aec9d08aa32d05ae2fb8647943f81a',
    name: 'HAIM',
    uri: 'spotify:artist:4Ui2kfOqGujY81UcPrb5KE',
  },
  {
    genres: [
      'chamber pop',
      'indie folk',
      'shimmer pop',
    ],
    id: '46CitWgnWrvF9t70C2p1Me',
    image: 'https://i.scdn.co/image/8b6e70aafad73059422920cbe528ecbcdff74018',
    name: 'Daughter',
    uri: 'spotify:artist:46CitWgnWrvF9t70C2p1Me',
  },
  {
    genres: [
      'alt-indie rock',
      'garage rock',
      'indie rock',
      'indietronica',
      'modern alternative rock',
      'modern rock',
    ],
    id: '3btzEQD6sugImIHPMRgkwV',
    image: 'https://i.scdn.co/image/f54c5e6b3919940e204406ef6846ee8e197f8217',
    name: 'Wolf Alice',
    uri: 'spotify:artist:3btzEQD6sugImIHPMRgkwV',
  },
];

export const topTracks = [
  {
    album: 'Gone Now',
    artist: 'Bleachers',
    id: '3ySU5vwQB33iGulwcUL9qQ',
    image: 'https://i.scdn.co/image/683a338f942640da90a27169e87c29440f20c4bb',
    title: 'Don\'t Take The Money',
    uri: 'spotify:track:3ySU5vwQB33iGulwcUL9qQ',
  },
  {
    album: 'Something To Tell You',
    artist: 'HAIM',
    id: '20QoFTI0fp2dNNexlHDhTe',
    image: 'https://i.scdn.co/image/9925a5f35b0a0e7b6f16e8f12c80cb5b4f015d57',
    title: 'Walking Away',
    uri: 'spotify:track:20QoFTI0fp2dNNexlHDhTe',
  },
  {
    album: 'Big Mess',
    artist: 'Grouplove',
    id: '1Tcd9flKIOxXWqR28aDP3U',
    image: 'https://i.scdn.co/image/781fa2da2c0dc8fc64f093dd753d4acfd6647db1',
    title: 'Don\'t Stop Making It Happen',
    uri: 'spotify:track:1Tcd9flKIOxXWqR28aDP3U',
  },
];

export const playlist = {
  name: 'Top Tracks',
  description: '',
  image: 'https://mosaic.scdn.co/640/129c6e4a622dc50c0802042780f92631a9dcc14056cc004f3ff47cff0ad8fcf05461730a193d8d8263b1ad734c51e06c7faee7fc95fd3abf0bbb9bbe6f4c08a3f6b73b062d6ea8384add3e3889814448',
  url: 'https://open.spotify.com/playlist/6yZesRjpu9OPKCM77N2EQt',
  tracks: topTracks,
};
