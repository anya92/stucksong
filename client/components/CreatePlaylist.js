import React, { Component } from 'react';
import { connect } from 'react-redux';
import { createPlaylist } from '../actions';
import Loadable from './HOC/Loadable';

import PlaylistForm from './PlaylistForm';
import Loader from '../styles/loader';

const AsyncError = Loadable({
  loader: () => import('./Error/ErrorComponent'),
});

const AsyncPlaylist = Loadable({
  loader: () => import('./Playlist'),
});

class CreatePlaylist extends Component {
  state = {
    name: `${this.props.auth.username}'s Top Tracks`,
    description: '',
    numberOfTracks: 25,
  }

  createPlaylist = (e) => {
    e.preventDefault();
    const { name, description, numberOfTracks } = this.state;
    this.props.createPlaylist(name, description, numberOfTracks);
  }

  handleChange = (type, value) => {
    this.setState({ [type]: value });
  }

  render() {
    const { pending, playlist, error } = this.props.playlist;
    if (error) return <AsyncError error={error} />;
    if (pending) return <Loader>Loading...</Loader>;
    if (!playlist) {
      return (
        <PlaylistForm
          name={this.state.name}
          description={this.state.description}
          numberOfTracks={this.state.numberOfTracks}
          handleChange={this.handleChange}
          createPlaylist={this.createPlaylist}
        />
      );
    }
    return (
      <AsyncPlaylist
        playlist={playlist}
        modal={this.state.modal}
        toggleModal={this.toggleModal}
      />
    );
  }
}

function mapStateToProps(state) {
  return {
    auth: state.auth,
    playlist: state.playlist,
  };
}

export default connect(mapStateToProps, { createPlaylist })(CreatePlaylist);
