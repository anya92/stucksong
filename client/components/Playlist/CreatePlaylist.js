import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  shape,
  oneOfType,
  bool,
  string,
  number,
  arrayOf,
  func,
} from 'prop-types';
import { createPlaylist } from '../../actions';
import Loadable from '../HOC/Loadable';

import PlaylistForm from './PlaylistForm';
import Loader from '../../styles/loader';

const AsyncError = Loadable({
  loader: () => import('../Error/ErrorComponent'),
});

const AsyncPlaylist = Loadable({
  loader: () => import('./Playlist'),
});

export class CreatePlaylist extends Component {
  state = {
    name: `${this.props.auth.username}'s Top Tracks`,
    description: '',
    numberOfTracks: 25,
  }

  createPlaylist = (e) => {
    e.preventDefault();
    const { name, description, numberOfTracks } = this.state;
    this.props.createPlaylist(name, description, numberOfTracks);
  }

  handleChange = (type, value) => {
    this.setState({ [type]: value });
  }

  render() {
    const { pending, playlist, error } = this.props.playlist;
    if (error) return <AsyncError error={error} />;
    if (pending) return <Loader>Loading...</Loader>;
    if (!playlist) {
      return (
        <PlaylistForm
          name={this.state.name}
          description={this.state.description}
          numberOfTracks={this.state.numberOfTracks}
          handleChange={this.handleChange}
          createPlaylist={this.createPlaylist}
        />
      );
    }
    return (
      <AsyncPlaylist
        playlist={playlist}
      />
    );
  }
}

CreatePlaylist.propTypes = {
  auth: shape({
    username: string,
  }).isRequired,
  playlist: shape({
    pending: bool.isRequired,
    error: oneOfType([bool, string]).isRequired,
    playlist: shape({
      name: string,
      description: string,
      image: string,
      numberOfTracks: number,
      tracks: arrayOf(shape({})),
    }),
  }).isRequired,
  createPlaylist: func.isRequired,
};

function mapStateToProps(state) {
  return {
    auth: state.auth,
    playlist: state.playlist,
  };
}

export default connect(mapStateToProps, { createPlaylist })(CreatePlaylist);
