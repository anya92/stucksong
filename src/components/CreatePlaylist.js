import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { createPlaylist } from '../actions';

class CreatePlaylist extends Component {
  constructor(props) {
    super(props);
  
    this.state = {
      name: `${this.props.auth.username}'s Top Tracks`,
      description: '',
      numberOfTracks: 50
    };
  }

  createPlaylist() {
    const { name, description, numberOfTracks } = this.state;
    this.props.createPlaylist(name, description, numberOfTracks);
  }

  renderContent() {
    switch(this.props.playlist) {
      case null: 
        return (
          <div>
            <div className="site-title site-title--visible">Create a playlist of your <Link to='/top-tracks'>Top&nbsp;Tracks</Link></div>
            <div className="playlist__form">
              <div className="form-group">
                <label htmlFor="name">Name</label>
                <div className="input-group">
                  <input 
                    type="text"
                    id="name"
                    maxLength="100"
                    className="form-control"
                    value={this.state.name}
                    onChange={e => this.setState({ name: e.target.value })}
                  />
                  <div className="input-group-addon">{this.state.name.length}/100</div>
                </div>
              </div>
              <div className="form-group">
                <label htmlFor="description">Description</label>
                <div className="input-group">
                  <textarea 
                    rows={3}
                    id="description"
                    maxLength="300"
                    className="form-control"
                    placeholder="Give your playlist a description."
                    value={this.state.description}
                    onChange={e => this.setState({ description: e.target.value })}
                  />
                  <div className="input-group-addon">{this.state.description.length}/300</div>
                </div>
              </div>
              <div className="form-group row">
                <label 
                  htmlFor="numberOfTracks"
                >
                  Number of Tracks
                </label>
                  <input 
                    type="number" 
                    id="numberOfTracks"
                    className={`form-control ${this.state.numberOfTracks < 10 || this.state.numberOfTracks > 50 ? 'input-danger' : ''}`}
                    min={10}
                    max={50}
                    value={this.state.numberOfTracks}
                    onChange={e => this.setState({ numberOfTracks: e.target.value })}
                  />
              </div>
              {(this.state.numberOfTracks < 10 || this.state.numberOfTracks > 50) && <div className="danger-info">Number of tracks must be between 10 and 50.</div>}
              <div className="button">
                <button className="btn" onClick={() => this.createPlaylist()}>Create</button>
              </div>
            </div>
          </div>
        );
      default: {
      const tracksPart1 = this.props.playlist.tracksImages.slice(0,5).map((url, i) => <div key={i}><img src={url} alt="" /></div>);
      const tracksPart2 = this.props.playlist.tracksImages.slice(5,10).map((url, i) => <div key={i + 5}><img src={url} alt="" /></div>);
        return (
          <div>
            <div className="playlist__collage">{ tracksPart1 }</div>  
            <div className="playlist__info">
              <div className="playlist__info__name">{this.props.playlist.name}</div>
              <div className="playlist__info__author">Created by {this.props.auth.username}</div>
              <div className="playlist__info__description">{this.props.playlist.description}</div>
              <div className="playlist__info__numberOfTracks"><span>{this.props.playlist.numberOfTracks}</span> Tracks</div>
            </div>
            <div className="playlist__collage">{ tracksPart2 }</div>
            <div className="playlist__link">
              <a href={this.props.playlist.url} target="_blank">Open Playlist</a>
            </div>
          </div>
        );
      }  	
    }
  }

  render() {
    return (
      <div className="playlist container">
        { this.renderContent() }
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    auth: state.auth,
    // playlist: state.playlist
    playlist: {
      name: 'heroldcwany\'s Top Tracks',
      description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
      url: 'http://github.com/anya92',
      numberOfTracks: 50,
      tracksImages: ['https://i.scdn.co/image/77d5914dde0b423a695195bf7c9af1510b202965', 'https://i.scdn.co/image/5a6ffa8082d067fa48ed5dbe4c4c528b9bb0a76a', 'https://i.scdn.co/image/c3eb558cabb62c9701215bb4f858c25edb22664e', 'https://i.scdn.co/image/ffcc57d638e9a2fda9bbff634faebc0bb50c6044', 'https://i.scdn.co/image/46c2987fad3ee33b137fc51e91791e5ddebf8e6e', 'https://i.scdn.co/image/88acfc8bbbd077673678638e2bc17e5b38f7e302', 'https://i.scdn.co/image/87ef475189ee1e7067e1e22c6be43bf11d8b7953', 'https://i.scdn.co/image/c389553d5d03cdf067e2624ca0e19d04d463f85e', 'https://i.scdn.co/image/d8296568ae1b856050976111fa892d8db693efd5', 'https://i.scdn.co/image/8358a9e995cc544ad3793acf3434ac3792783f44', 'https://i.scdn.co/image/77d5914dde0b423a695195bf7c9af1510b202965', 'https://i.scdn.co/image/5a6ffa8082d067fa48ed5dbe4c4c528b9bb0a76a', 'https://i.scdn.co/image/c3eb558cabb62c9701215bb4f858c25edb22664e', 'https://i.scdn.co/image/ffcc57d638e9a2fda9bbff634faebc0bb50c6044', 'https://i.scdn.co/image/46c2987fad3ee33b137fc51e91791e5ddebf8e6e', 'https://i.scdn.co/image/88acfc8bbbd077673678638e2bc17e5b38f7e302', 'https://i.scdn.co/image/87ef475189ee1e7067e1e22c6be43bf11d8b7953', 'https://i.scdn.co/image/c389553d5d03cdf067e2624ca0e19d04d463f85e', 'https://i.scdn.co/image/d8296568ae1b856050976111fa892d8db693efd5', 'https://i.scdn.co/image/8358a9e995cc544ad3793acf3434ac3792783f44'],
    }
  };
}

export default connect(mapStateToProps, { createPlaylist })(CreatePlaylist);
