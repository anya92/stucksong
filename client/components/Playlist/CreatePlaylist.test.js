import React from 'react';
import { shallow } from 'enzyme';

import { CreatePlaylist } from './CreatePlaylist';
import { playlist } from '../../assets/sampleData';

describe('CreatePlaylist component', () => {
  let wrapper;
  let createPlaylist;
  beforeEach(() => {
    createPlaylist = jest.fn();
    const mockProps = {
      auth: {
        username: 'user',
      },
      playlist: {
        playlist: null,
        pending: false,
        error: false,
      },
      createPlaylist,
    };
    wrapper = shallow(<CreatePlaylist {...mockProps} />);
  });

  it('renders PlaylistForm correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Playlist component correctly', () => {
    wrapper.setProps({
      playlist: {
        playlist,
        pending: false,
        error: false,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with pending correctly', () => {
    wrapper.setProps({
      playlist: {
        playlist: null,
        pending: true,
        error: false,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with error correctly', () => {
    wrapper.setProps({
      playlist: {
        playlist: null,
        pending: false,
        error: 'Request failed',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
