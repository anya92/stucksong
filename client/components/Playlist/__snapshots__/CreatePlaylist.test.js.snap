// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePlaylist component renders Playlist component correctly 1`] = `
<LoadableComponent
  playlist={
    Object {
      "description": "",
      "image": "https://mosaic.scdn.co/640/129c6e4a622dc50c0802042780f92631a9dcc14056cc004f3ff47cff0ad8fcf05461730a193d8d8263b1ad734c51e06c7faee7fc95fd3abf0bbb9bbe6f4c08a3f6b73b062d6ea8384add3e3889814448",
      "name": "Top Tracks",
      "tracks": Array [
        Object {
          "album": "Gone Now",
          "artist": "Bleachers",
          "id": "3ySU5vwQB33iGulwcUL9qQ",
          "image": "https://i.scdn.co/image/683a338f942640da90a27169e87c29440f20c4bb",
          "title": "Don't Take The Money",
          "uri": "spotify:track:3ySU5vwQB33iGulwcUL9qQ",
        },
        Object {
          "album": "Something To Tell You",
          "artist": "HAIM",
          "id": "20QoFTI0fp2dNNexlHDhTe",
          "image": "https://i.scdn.co/image/9925a5f35b0a0e7b6f16e8f12c80cb5b4f015d57",
          "title": "Walking Away",
          "uri": "spotify:track:20QoFTI0fp2dNNexlHDhTe",
        },
        Object {
          "album": "Big Mess",
          "artist": "Grouplove",
          "id": "1Tcd9flKIOxXWqR28aDP3U",
          "image": "https://i.scdn.co/image/781fa2da2c0dc8fc64f093dd753d4acfd6647db1",
          "title": "Don't Stop Making It Happen",
          "uri": "spotify:track:1Tcd9flKIOxXWqR28aDP3U",
        },
      ],
      "url": "https://open.spotify.com/playlist/6yZesRjpu9OPKCM77N2EQt",
    }
  }
/>
`;

exports[`CreatePlaylist component renders PlaylistForm correctly 1`] = `
<PlaylistForm
  createPlaylist={[Function]}
  description=""
  handleChange={[Function]}
  name="user's Top Tracks"
  numberOfTracks={25}
/>
`;

exports[`CreatePlaylist component renders with error correctly 1`] = `
<LoadableComponent
  error="Request failed"
/>
`;

exports[`CreatePlaylist component renders with pending correctly 1`] = `
<glamorous(div)>
  Loading...
</glamorous(div)>
`;
